tests/cover/test_deferred_strategies.py

```diff
@@ -33,27 +33,13 @@ def test_binary_tree():
     assert minimal(tree, lambda x: isinstance(x, tuple)) == (0, 0)
 
 
-def test_bad_binary_tree():
```

Comment:
The new generation strategy was triggering bad behaviour in them, which is kinda expected. These strategies have two major problems:

1. When generated uniformly at random, their expected size is infinite
2. They do exactly the wrong thing when the size bounding kicks in - we try to replace tails with 0 to trigger shrinking, but when you do that with these strategies it will do the opposite - it will cause them to draw more data!

Because in this change we generate a uniform(ish) initial prefix, it increases the expected size significantly because that gives us a large chunk of branch nodes and we have to draw children for all of them. This means we tend to actually hit that limit and everything blows up. Previously we happened to do a decent job of generating these (I'm not sure why).

Anyway, because of (2) the health check introduced in #996 would fail this strategy if you actually tried to use it, so I decided I didn't care to try to make this work.

Reply:
👍 we even have docs that you shouldn't try to write strategies like that!